### ISSUE TITLE
HTTPS/Adsense

### ISSUE THREAD

**Issue by hackmethod** [google/ctfscoreboard#126]
What needs to be edited to make this scoreboard HTTPS?  I have been attempting to make an instance of this HTTPS and add google analytics and adsense to it with no success.  Any help would be greatly appreciated.  Thanks in advance.

**Comment by k9delight** [comment]
HTTPS should be straight forward, if you're using appengine, setting
secure: always should do the trick, along with some config modifications.

As for adsense, It's likely you'll need to add a custom CSP policy so that
adsense can load javascript / do whatever it does.

Take a look at

https://github.com/google/ctfscoreboard/blob/013018c3685cc6a42daf5c6b81c42c45ebb39264/scoreboard/config_defaults.py#L21

and

https://github.com/google/ctfscoreboard/blob/14b276213c9aa628d26f3da56a42fe62460eb619/scoreboard/context.py#L28

and that should give you an idea of what you need to do. You'll probably
want to enable developer tools in your browser to see what is causing
conflicts.

**Comment by hackmethod** [comment]
I think I may have been editing the wrong nginx.conf (not the doc/nginx.conf) when attempting HTTPS, I will revisit this issue.

The CSP is very new to me and I am not entirely certain what I am looking at when those lines.  I did run developer tools and just noticed the "blocked" portion on any adsense or google metrics I had added to the base page.  Any chance you have a tutorial on writing a custom CSP policy before I start googling my heart out?
